platform/android/MapboxGLAndroidSDK/src/cpp/map_renderer.cpp

```diff
@@ -84,6 +84,20 @@ void MapRenderer::schedule(std::function<void()>&& scheduled) {
     }
 }
 
+std::size_t MapRenderer::waitForEmpty(Milliseconds timeout) {
+    try {
+        android::UniqueEnv _env = android::AttachEnv();
+        static auto& javaClass = jni::Class<MapRenderer>::Singleton(*_env);
+        static auto waitForEmpty = javaClass.GetMethod<jni::jint(jni::jlong)>(*_env, "waitForEmpty");
+        if (auto weakReference = javaPeer.get(*_env)) {
+            return weakReference.Call(*_env, waitForEmpty, static_cast<int64_t>(timeout.count()));
+        }
+    } catch (const std::exception& exception) {
+        Log::Error(Event::Android, std::string("MapRenderer::waitForEmpty failed: ") + exception.what());
+    }
+    return -1;
```

Comment:
This will underflow to `std::numeric_limits<size_t>::max()`., If intentional, return that instead of allowing underflow.

Reply:
Good catch, there really isn't a correct value there.  An exception would be more appropriate to the Java caller, I guess.

Reply:
If I understand the JNI stuff correctly the else condition there isn't really an error, but shouldn't happen on a live object.  Nothing else appears to throw in that case, so returning zero (i.e., nothing still pending) seems appropriate.

Other methods there aren't translating exceptions, though there's a facility for doing so, but that seems appropriate.